rust/src/pet/participant.rs

```diff
@@ -206,12 +220,12 @@ impl<'tag, 'cert, 'sign_, 'url, 'dict> UpdateBoxBuffer<'tag, 'cert, 'sign_, 'url
         dict_seed: &'dict [u8],
     ) -> Self {
         Self(
-            b"update",               // 6 bytes
-            certificate,             // 0 bytes (dummy)
-            &signature_sum.0[..],    // 64 bytes
-            &signature_update.0[..], // 64 bytes
-            model_url,               // 32 bytes (dummy)
-            dict_seed,               // 112 * dict_sum.len() bytes
+            b"update",                 // 6 bytes
+            certificate,               // 0 bytes (dummy)
+            signature_sum.as_ref(),    // 64 bytes
+            signature_update.as_ref(), // 64 bytes
+            model_url,                 // 32 bytes (dummy)
```

Comment:
This is fine as a very short term solution, but we should define a proper encoding for variable length fields. A simple way to do that is to use TLV encoding. I guess certificates have a variable length as well? If so we should put it at the end of the message, with the other variable length fields.

Reply:
the `dummy` values are just placeholders for now, as soon as they become meaningful it definitely makes sense to encode them properly. certificates are most likely going to be variable in size. regarding the full message length, it also depends on if we are going to use padding for a constant message length (for attack vector reasons) or not.